src/Compilers/Core/Portable/Symbols/Attributes/CommonParameterEarlyWellKnownAttributeData.cs

```diff
@@ -80,6 +80,38 @@ public bool HasCallerMemberNameAttribute
                 SetDataStored();
             }
         }
+
+        private bool _hasCallerArgumentExpressionAttribute;
```

Comment:
>_hasCallerArgumentExpressionAttribute [](start = 21, length = 37)

Why do we need this boolean? Isn't _argumentExpressionParameterIndex sufficient?

Reply:
@AlekseyTs I attempt to get the index based on the value of `HasCallerArgumentExpressionAttribute`.

Reply:
>> Why do we need this boolean? Isn't _argumentExpressionParameterIndex sufficient?

> I attempt to get the index based on the value of HasCallerArgumentExpressionAttribute.

I can see that, but it looks like we don't really have to do that.

---
In reply to: [600237904](https://github.com/dotnet/roslyn/pull/51952#discussion_r600237904) [](ancestors = 600237904)

Reply:
Done.